File: Python/case_utils/case_utils_test.py
"""
Case Utils 测试文件

测试命名风格转换工具的所有功能。
"""

import unittest
from mod import CaseUtils, to_camel_case, to_pascal_case, to_snake_case, to_kebab_case, detect_case, convert_case


class TestCaseUtils(unittest.TestCase):
    """CaseUtils 类测试"""
    
    def test_split_words(self):
        """测试单词拆分"""
        self.assertEqual(CaseUtils.split_words("helloWorld"), ['hello', 'World'])
        self.assertEqual(CaseUtils.split_words("hello_world"), ['hello', 'world'])
        self.assertEqual(CaseUtils.split_words("HelloWorld"), ['Hello', 'World'])
        self.assertEqual(CaseUtils.split_words("HELLO_WORLD"), ['HELLO', 'WORLD'])
        self.assertEqual(CaseUtils.split_words("hello-world"), ['hello', 'world'])
        self.assertEqual(CaseUtils.split_words("hello.world"), ['hello', 'world'])
        self.assertEqual(CaseUtils.split_words("hello world"), ['hello', 'world'])
        self.assertEqual(CaseUtils.split_words("HelloWorld123"), ['Hello', 'World', '123'])
        self.assertEqual(CaseUtils.split_words("XMLHttpRequest"), ['XML', 'Http', 'Request'])
        self.assertEqual(CaseUtils.split_words(""), [])
    
    def test_to_camel_case(self):
        """测试转换为 camelCase"""
        self.assertEqual(CaseUtils.to_camel_case("hello_world"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("HelloWorld"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("HELLO_WORLD"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("hello-world"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("hello.world"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("Hello World"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("xml_http_request"), "xmlHttpRequest")
        self.assertEqual(CaseUtils.to_camel_case(""), "")
    
    def test_to_pascal_case(self):
        """测试转换为 PascalCase"""
        self.assertEqual(CaseUtils.to_pascal_case("hello_world"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("helloWorld"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("HELLO_WORLD"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("hello-world"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("hello.world"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("Hello World"), "HelloWorld")
        self.assertEqual(CaseUtils.to_pascal_case("xml_http_request"), "XmlHttpRequest")
        self.assertEqual(CaseUtils.to_pascal_case(""), "")
    
    def test_to_snake_case(self):
        """测试转换为 snake_case"""
        self.assertEqual(CaseUtils.to_snake_case("helloWorld"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("HelloWorld"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("HELLO_WORLD"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("hello-world"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("hello.world"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("Hello World"), "hello_world")
        self.assertEqual(CaseUtils.to_snake_case("xmlHttpRequest"), "xml_http_request")
        self.assertEqual(CaseUtils.to_snake_case(""), "")
    
    def test_to_screaming_snake_case(self):
        """测试转换为 SCREAMING_SNAKE_CASE"""
        self.assertEqual(CaseUtils.to_screaming_snake_case("helloWorld"), "HELLO_WORLD")
        self.assertEqual(CaseUtils.to_screaming_snake_case("HelloWorld"), "HELLO_WORLD")
        self.assertEqual(CaseUtils.to_screaming_snake_case("hello_world"), "HELLO_WORLD")
        self.assertEqual(CaseUtils.to_screaming_snake_case("hello-world"), "HELLO_WORLD")
        self.assertEqual(CaseUtils.to_screaming_snake_case("hello.world"), "HELLO_WORLD")
        self.assertEqual(CaseUtils.to_screaming_snake_case(""), "")
    
    def test_to_kebab_case(self):
        """测试转换为 kebab-case"""
        self.assertEqual(CaseUtils.to_kebab_case("helloWorld"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case("HelloWorld"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case("hello_world"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case("HELLO_WORLD"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case("hello.world"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case("Hello World"), "hello-world")
        self.assertEqual(CaseUtils.to_kebab_case(""), "")
    
    def test_to_train_case(self):
        """测试转换为 Train-Case"""
        self.assertEqual(CaseUtils.to_train_case("helloWorld"), "Hello-World")
        self.assertEqual(CaseUtils.to_train_case("hello_world"), "Hello-World")
        self.assertEqual(CaseUtils.to_train_case("HELLO_WORLD"), "Hello-World")
        self.assertEqual(CaseUtils.to_train_case(""), "")
    
    def test_to_dot_case(self):
        """测试转换为 dot.case"""
        self.assertEqual(CaseUtils.to_dot_case("helloWorld"), "hello.world")
        self.assertEqual(CaseUtils.to_dot_case("hello_world"), "hello.world")
        self.assertEqual(CaseUtils.to_dot_case("hello-world"), "hello.world")
        self.assertEqual(CaseUtils.to_dot_case(""), "")
    
    def test_to_title_case(self):
        """测试转换为 Title Case"""
        self.assertEqual(CaseUtils.to_title_case("helloWorld"), "Hello World")
        self.assertEqual(CaseUtils.to_title_case("hello_world"), "Hello World")
        self.assertEqual(CaseUtils.to_title_case("hello-world"), "Hello World")
        self.assertEqual(CaseUtils.to_title_case("HELLO_WORLD"), "Hello World")
        self.assertEqual(CaseUtils.to_title_case(""), "")
    
    def test_to_sentence_case(self):
        """测试转换为 Sentence case"""
        self.assertEqual(CaseUtils.to_sentence_case("helloWorld"), "Hello world")
        self.assertEqual(CaseUtils.to_sentence_case("HELLO_WORLD"), "Hello world")
        self.assertEqual(CaseUtils.to_sentence_case("hello_world"), "Hello world")
        self.assertEqual(CaseUtils.to_sentence_case(""), "")
    
    def test_to_path_case(self):
        """测试转换为 path/case"""
        self.assertEqual(CaseUtils.to_path_case("helloWorld"), "hello/world")
        self.assertEqual(CaseUtils.to_path_case("hello_world"), "hello/world")
        self.assertEqual(CaseUtils.to_path_case("hello-world"), "hello/world")
        self.assertEqual(CaseUtils.to_path_case(""), "")
    
    def test_detect_case(self):
        """测试命名风格检测"""
        self.assertEqual(CaseUtils.detect_case("helloWorld"), "camel")
        self.assertEqual(CaseUtils.detect_case("HelloWorld"), "pascal")
        self.assertEqual(CaseUtils.detect_case("hello_world"), "snake")
        self.assertEqual(CaseUtils.detect_case("HELLO_WORLD"), "screaming_snake")
        self.assertEqual(CaseUtils.detect_case("hello-world"), "kebab")
        self.assertEqual(CaseUtils.detect_case("Hello-World"), "train")
        self.assertEqual(CaseUtils.detect_case("hello.world"), "dot")
        self.assertEqual(CaseUtils.detect_case("Hello World"), "title")
        self.assertEqual(CaseUtils.detect_case("hello"), "unknown")
        self.assertIsNone(CaseUtils.detect_case(""))
    
    def test_convert(self):
        """测试通用转换方法"""
        self.assertEqual(CaseUtils.convert("hello_world", "camel"), "helloWorld")
        self.assertEqual(CaseUtils.convert("hello_world", "camelCase"), "helloWorld")
        self.assertEqual(CaseUtils.convert("helloWorld", "snake"), "hello_world")
        self.assertEqual(CaseUtils.convert("helloWorld", "snake_case"), "hello_world")
        self.assertEqual(CaseUtils.convert("hello_world", "pascal"), "HelloWorld")
        self.assertEqual(CaseUtils.convert("hello_world", "kebab"), "hello-world")
        self.assertEqual(CaseUtils.convert("hello_world", "kebab-case"), "hello-world")
        self.assertEqual(CaseUtils.convert("hello_world", "dot"), "hello.world")
        self.assertEqual(CaseUtils.convert("hello_world", "title"), "Hello World")
        
        with self.assertRaises(ValueError):
            CaseUtils.convert("hello", "invalid_case")
    
    def test_convert_all(self):
        """测试转换为所有格式"""
        result = CaseUtils.convert_all("helloWorld")
        
        self.assertEqual(result['camelCase'], "helloWorld")
        self.assertEqual(result['PascalCase'], "HelloWorld")
        self.assertEqual(result['snake_case'], "hello_world")
        self.assertEqual(result['SCREAMING_SNAKE_CASE'], "HELLO_WORLD")
        self.assertEqual(result['kebab-case'], "hello-world")
        self.assertEqual(result['Train-Case'], "Hello-World")
        self.assertEqual(result['dot.case'], "hello.world")
        self.assertEqual(result['Title Case'], "Hello World")
        self.assertEqual(result['Sentence case'], "Hello world")
        self.assertEqual(result['path/case'], "hello/world")
    
    def test_is_valid_identifier(self):
        """测试命名风格验证"""
        self.assertTrue(CaseUtils.is_valid_identifier("helloWorld", "camel"))
        self.assertFalse(CaseUtils.is_valid_identifier("hello_world", "camel"))
        self.assertTrue(CaseUtils.is_valid_identifier("HelloWorld", "pascal"))
        self.assertFalse(CaseUtils.is_valid_identifier("helloWorld", "pascal"))
        self.assertTrue(CaseUtils.is_valid_identifier("hello_world", "snake"))
        self.assertTrue(CaseUtils.is_valid_identifier("HELLO_WORLD", "screaming_snake"))
        self.assertTrue(CaseUtils.is_valid_identifier("hello-world", "kebab"))
    
    def test_batch_convert(self):
        """测试批量转换"""
        inputs = ["hello_world", "fooBar", "BazQux"]
        results = CaseUtils.batch_convert(inputs, "camel")
        
        self.assertEqual(results, ["helloWorld", "fooBar", "bazQux"])
    
    def test_to_plural_snake(self):
        """测试复数 snake_case"""
        self.assertEqual(CaseUtils.to_plural_snake("userAccount"), "user_accounts")
        self.assertEqual(CaseUtils.to_plural_snake("category"), "categories")
        self.assertEqual(CaseUtils.to_plural_snake("box"), "boxes")
        self.assertEqual(CaseUtils.to_plural_snake("item"), "items")
        self.assertEqual(CaseUtils.to_plural_snake(""), "")


class TestConvenienceFunctions(unittest.TestCase):
    """测试便捷函数"""
    
    def test_to_camel_case_function(self):
        self.assertEqual(to_camel_case("hello_world"), "helloWorld")
    
    def test_to_pascal_case_function(self):
        self.assertEqual(to_pascal_case("hello_world"), "HelloWorld")
    
    def test_to_snake_case_function(self):
        self.assertEqual(to_snake_case("helloWorld"), "hello_world")
    
    def test_to_kebab_case_function(self):
        self.assertEqual(to_kebab_case("helloWorld"), "hello-world")
    
    def test_detect_case_function(self):
        self.assertEqual(detect_case("helloWorld"), "camel")
    
    def test_convert_case_function(self):
        self.assertEqual(convert_case("hello_world", "camel"), "helloWorld")


class TestEdgeCases(unittest.TestCase):
    """测试边界情况"""
    
    def test_empty_string(self):
        """测试空字符串"""
        self.assertEqual(CaseUtils.to_camel_case(""), "")
        self.assertEqual(CaseUtils.to_snake_case(""), "")
        self.assertEqual(CaseUtils.split_words(""), [])
    
    def test_single_word(self):
        """测试单个单词"""
        self.assertEqual(CaseUtils.to_camel_case("hello"), "hello")
        self.assertEqual(CaseUtils.to_pascal_case("hello"), "Hello")
        self.assertEqual(CaseUtils.to_snake_case("hello"), "hello")
    
    def test_numbers(self):
        """测试包含数字"""
        self.assertEqual(CaseUtils.to_snake_case("version2Update"), "version_2_update")
        self.assertEqual(CaseUtils.to_camel_case("version_2_update"), "version2Update")
    
    def test_consecutive_uppercase(self):
        """测试连续大写字母"""
        self.assertEqual(CaseUtils.to_snake_case("XMLHttpRequest"), "xml_http_request")
        self.assertEqual(CaseUtils.to_snake_case("IOError"), "io_error")
        self.assertEqual(CaseUtils.to_camel_case("XML_HTTP_REQUEST"), "xmlHttpRequest")
    
    def test_multiple_separators(self):
        """测试多个分隔符"""
        self.assertEqual(CaseUtils.to_camel_case("hello__world"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("hello--world"), "helloWorld")
        self.assertEqual(CaseUtils.to_camel_case("hello..world"), "helloWorld")
    
    def test_mixed_separators(self):
        """测试混合分隔符"""
        self.assertEqual(CaseUtils.to_snake_case("hello-world.test"), "hello_world_test")
        self.assertEqual(CaseUtils.to_kebab_case("hello_world.test"), "hello-world-test")


if __name__ == "__main__":
    unittest.main(verbosity=2)

File: Go/case_utils/case_utils_test.go
package case_utils

import (
	"testing"
)

func TestToCamelCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"hello_world", "helloWorld"},
		{"HelloWorld", "helloWorld"},
		{"hello-world", "helloWorld"},
		{"HELLO_WORLD", "helloWorld"},
		{"hello", "hello"},
		{"Hello", "hello"},
		{"helloWorld", "helloWorld"},
		{"hello_world_test", "helloWorldTest"},
		{"HTMLParser", "htmlParser"},
		{"parseHTMLResponse", "parseHtmlResponse"},
		{"XMLHttpRequest", "xmlHttpRequest"},
		{"kebab-case-example", "kebabCaseExample"},
		{"dot.case.example", "dotCaseExample"},
		{"path/case/example", "pathCaseExample"},
		{"APIEndpoint", "apiEndpoint"},
		{"getUserID", "getUserId"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToCamelCase(test.input)
		if result != test.expected {
			t.Errorf("ToCamelCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToPascalCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"hello_world", "HelloWorld"},
		{"hello-world", "HelloWorld"},
		{"HELLO_WORLD", "HelloWorld"},
		{"hello", "Hello"},
		{"helloWorld", "HelloWorld"},
		{"hello_world_test", "HelloWorldTest"},
		{"HTMLParser", "HtmlParser"},
		{"parseHTMLResponse", "ParseHtmlResponse"},
		{"kebab-case-example", "KebabCaseExample"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToPascalCase(test.input)
		if result != test.expected {
			t.Errorf("ToPascalCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToSnakeCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "hello_world"},
		{"HelloWorld", "hello_world"},
		{"hello-world", "hello_world"},
		{"HELLO_WORLD", "hello_world"},
		{"hello", "hello"},
		{"Hello", "hello"},
		{"helloWorldTest", "hello_world_test"},
		{"HTMLParser", "html_parser"},
		{"parseHTMLResponse", "parse_html_response"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToSnakeCase(test.input)
		if result != test.expected {
			t.Errorf("ToSnakeCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToKebabCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "hello-world"},
		{"HelloWorld", "hello-world"},
		{"hello_world", "hello-world"},
		{"HELLO_WORLD", "hello-world"},
		{"hello", "hello"},
		{"helloWorldTest", "hello-world-test"},
		{"HTMLParser", "html-parser"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToKebabCase(test.input)
		if result != test.expected {
			t.Errorf("ToKebabCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToScreamingSnakeCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "HELLO_WORLD"},
		{"HelloWorld", "HELLO_WORLD"},
		{"hello_world", "HELLO_WORLD"},
		{"hello-world", "HELLO_WORLD"},
		{"hello", "HELLO"},
		{"helloWorldTest", "HELLO_WORLD_TEST"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToScreamingSnakeCase(test.input)
		if result != test.expected {
			t.Errorf("ToScreamingSnakeCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToUpperCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "HELLOWORLD"},
		{"HelloWorld", "HELLOWORLD"},
		{"hello_world", "HELLOWORLD"},
		{"hello", "HELLO"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToUpperCase(test.input)
		if result != test.expected {
			t.Errorf("ToUpperCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToLowerCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"HelloWorld", "helloworld"},
		{"HELLO_WORLD", "helloworld"},
		{"hello_world", "helloworld"},
		{"HELLO", "hello"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToLowerCase(test.input)
		if result != test.expected {
			t.Errorf("ToLowerCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToTitleCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"hello_world", "Hello World"},
		{"helloWorld", "Hello World"},
		{"HELLO_WORLD", "Hello World"},
		{"hello-world", "Hello World"},
		{"hello", "Hello"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToTitleCase(test.input)
		if result != test.expected {
			t.Errorf("ToTitleCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToSentenceCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"hello_world", "Hello world"},
		{"helloWorld", "Hello world"},
		{"HELLO_WORLD", "Hello world"},
		{"hello-world", "Hello world"},
		{"hello", "Hello"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToSentenceCase(test.input)
		if result != test.expected {
			t.Errorf("ToSentenceCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToDotCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "hello.world"},
		{"HelloWorld", "hello.world"},
		{"hello_world", "hello.world"},
		{"hello-world", "hello.world"},
		{"helloWorldTest", "hello.world.test"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToDotCase(test.input)
		if result != test.expected {
			t.Errorf("ToDotCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToPathCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "hello/world"},
		{"HelloWorld", "hello/world"},
		{"hello_world", "hello/world"},
		{"hello-world", "hello/world"},
		{"helloWorldTest", "hello/world/test"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToPathCase(test.input)
		if result != test.expected {
			t.Errorf("ToPathCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestToTrainCase(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"helloWorld", "Hello-World"},
		{"HelloWorld", "Hello-World"},
		{"hello_world", "Hello-World"},
		{"hello-world", "Hello-World"},
		{"helloWorldTest", "Hello-World-Test"},
		{"", ""},
	}

	for _, test := range tests {
		result := ToTrainCase(test.input)
		if result != test.expected {
			t.Errorf("ToTrainCase(%q) = %q, expected %q", test.input, result, test.expected)
		}
	}
}

func TestDetectCase(t *testing.T) {
	tests := []struct {
		input    string
		expected CaseType
	}{
		{"helloWorld", CaseCamel},
		{"HelloWorld", CasePascal},
		{"hello_world", CaseSnake},
		{"hello-world", CaseKebab},
		{"HELLO_WORLD", CaseScreamingSnake},
		{"HELLO", CaseUpper},
		{"hello", CaseLower},
		{"Hello World", CaseTitle},
		{"", CaseUnknown},
	}

	for _, test := range tests {
		result := DetectCase(test.input)
		if result != test.expected {
			t.Errorf("DetectCase(%q) = %v, expected %v", test.input, result, test.expected)
		}
	}
}

func TestConvert(t *testing.T) {
	tests := []struct {
		input    string
		to       CaseType
		expected string
	}{
		{"hello_world", CaseCamel, "helloWorld"},
		{"hello_world", CasePascal, "HelloWorld"},
		{"HelloWorld", CaseSnake, "hello_world"},
		{"HelloWorld", CaseKebab, "hello-world"},
		{"helloWorld", CaseScreamingSnake, "HELLO_WORLD"},
		{"helloWorld", CaseUpper, "HELLOWORLD"},
		{"HelloWorld", CaseLower, "helloworld"},
		{"hello_world", CaseTitle, "Hello World"},
	}

	for _, test := range tests {
		result := Convert(test.input, test.to)
		if result != test.expected {
			t.Errorf("Convert(%q, %v) = %q, expected %q", test.input, test.to, result, test.expected)
		}
	}
}

func TestCaseTypeString(t *testing.T) {
	tests := []struct {
		caseType CaseType
		expected string
	}{
		{CaseCamel, "camelCase"},
		{CasePascal, "PascalCase"},
		{CaseSnake, "snake_case"},
		{CaseKebab, "kebab-case"},
		{CaseScreamingSnake, "SCREAMING_SNAKE_CASE"},
		{CaseUpper, "UPPERCASE"},
		{CaseLower, "lowercase"},
		{CaseTitle, "Title Case"},
		{CaseUnknown, "unknown"},
	}

	for _, test := range tests {
		result := test.caseType.String()
		if result != test.expected {
			t.Errorf("CaseType(%d).String() = %q, expected %q", test.caseType, result, test.expected)
		}
	}
}

func TestToConstantCase(t *testing.T) {
	// ToConstantCase should be same as ToScreamingSnakeCase
	input := "helloWorld"
	if ToConstantCase(input) != ToScreamingSnakeCase(input) {
		t.Errorf("ToConstantCase should equal ToScreamingSnakeCase")
	}
}

// Benchmark tests
func BenchmarkToCamelCase(b *testing.B) {
	for i := 0; i < b.N; i++ {
		ToCamelCase("hello_world_test_case")
	}
}

func BenchmarkToSnakeCase(b *testing.B) {
	for i := 0; i < b.N; i++ {
		ToSnakeCase("helloWorldTestCase")
	}
}

func BenchmarkDetectCase(b *testing.B) {
	for i := 0; i < b.N; i++ {
		DetectCase("helloWorldTestCase")
	}
}